meta.go

```diff
@@ -46,6 +47,38 @@ const (
 	RolePeer     RelationRole = "peer"
 )
 
+// Feature defines a piece of functionality provided by a Juju controller.
+type Feature string
+
+// Supported feature types.
+const (
+	FeatureContainers Feature = "containers"
```

Comment:
Currently we unmarshal the `assumes` field into a list of strings as we do with all other metadata fields. I don't think there would be any benefit in treating this as a blob that we store to the database and parse on demand since that would mean that we simply defer the interpretation of the field to a later stage. 

Also, note that the `assumes` payload is part of a YAML document and given the flexibility of YAML syntax, parsing it (manually)( would be a non trivial task.

We still must figure out what are the semantics of working with a parsed assumes value (is it and, is it or? etc.) and I would prefer that we don't implement several potential variants (the equivalent of versioning) for a brand new feature that no charm out there actively uses (yet). Instead, I believe the focus should be on coming up with an actual concrete spec for `assumes` that covers all the use cases that we eventually want to support and implement that.